### PR TITLE
Issue/2112 generateNormalizeStyles() partial

### PR DESCRIPTION
**Pull request by tatjana-gartner2**
### Pull Request Checklist

- [x] I have read and accept the [Contributor License Agreement](https://github.com/porscheofficial/oss-docs/blob/67c5b24838a293ce7a964884e6005eb71f2b8579/CONTRIBUTOR_LICENSE_AGREEMENT.md)

#### References

- Preview: https://designsystem.porsche.com/issue/.../

#### Scope

Short description of the pull request scope

### CODE REVIEW

File: packages/components-angular/src/app/app.component.ts [hunk 1]
```diff
@@ -5,7 +5,7 @@ import { routes } from './app-routing.module';
 @Component({
   selector: 'app-root',
   template: `
-    <select (change)="changeRoute($event.target.value)">
+    <select (change)="changeRoute($event.target.value)" style="font: revert; overflow-wrap: revert; hyphens: revert">
```
Comment (tatjana-gartner2) [inline packages/components-angular/src/app/app.component.ts:8]: move to shared package

File: packages/components-js/projects/partials/scripts/generateInitialStylesPartial.ts [hunk 2]
```diff
@@ -8,19 +14,104 @@ export const generateInitialStylesPartial = (): string => {
...
 
+  const normalizeStyles: Styles = {
+    '@global': {
+      html: {
+        textSizeAdjust: 'none',
+        WebkitTextSizeAdjust: 'none', // stop iOS safari from adjusting font size when screen rotation is changing
+        fontFamily: font.family,
+      },
+
+      'button, input, optgroup, select, textarea': {
```
Comment (tatjana-gartner2) [inline packages/components-js/projects/partials/scripts/generateInitialStylesPartial.ts:31]: try * selector

File: packages/components-js/projects/partials/scripts/generateInitialStylesPartial.ts [hunk 3]
```diff
@@ -1,5 +1,11 @@
 import { joinArrayElementsToString, withoutTagsOption } from './utils';
 import { INTERNAL_TAG_NAMES, TAG_NAMES } from '@porsche-design-system/shared';
+import { create, Styles } from 'jss';
+import { color, font } from '@porsche-design-system/utilities';
```
Comment (tatjana-gartner2) [inline packages/components-js/projects/partials/scripts/generateInitialStylesPartial.ts:4]: utilities/v2

File: packages/components-js/projects/partials/scripts/generateInitialStylesPartial.ts [hunk 4]
```diff
@@ -8,19 +14,104 @@ export const generateInitialStylesPartial = (): string => {
...
+        textSizeAdjust: 'none',
+        WebkitTextSizeAdjust: 'none', // stop iOS safari from adjusting font size when screen rotation is changing
+        fontFamily: font.family,
+        fontSize: font.size.small.fontSize,
+        lineHeight: font.size.small.lineHeight,
+        fontWeight: font.weight.regular,
+        fontStyle: 'normal',
+        fontVariant: 'normal',
+        overflowWrap: 'break-word',
+        hyphens: 'auto',
```
Comment (tatjana-gartner2) [inline packages/components-js/projects/partials/scripts/generateInitialStylesPartial.ts:41]: if * selctor without h1-h6

File: packages/components-js/projects/partials/scripts/generateInitialStylesPartial.ts [hunk 5]
```diff
@@ -8,19 +14,104 @@ export const generateInitialStylesPartial = (): string => {
...
+      },
+
+      'em, i': {
+        fontStyle: 'normal',
+      },
+
+      'a, button, input, select, textarea': {
+        outline: '1px solid transparent',
+        outlineOffset: '1px',
+        '&::-moz-focus-inner': {
```
Comment (tatjana-gartner2) [inline packages/components-js/projects/partials/scripts/generateInitialStylesPartial.ts:74]: remove focus-inner

File: packages/components-js/projects/partials/scripts/generateInitialStylesPartial.ts [hunk 6]
```diff
@@ -8,19 +14,104 @@ export const generateInitialStylesPartial = (): string => {
...
+          outlineColor: 'currentColor',
+        },
+        '&:focus:not(:focus-visible)': {
+          outlineColor: 'transparent',
+        },
+      },
+    },
+  };
+
+  const minifyCss = (css: string): string =>
```
Comment (tatjana-gartner2) [inline packages/components-js/projects/partials/scripts/generateInitialStylesPartial.ts:87]: import from shared package

File: packages/components-js/projects/partials/scripts/generateInitialStylesPartial.ts [hunk 7]
```diff
@@ -8,19 +14,104 @@ export const generateInitialStylesPartial = (): string => {
...
+        generateId: (rule) => rule.key,
+      })
+      .toString();
+
   const initialStylesFunction = `export function getInitialStyles(opts?: GetInitialStylesOptionsFormatHtml): string;
 export function getInitialStyles(opts?: GetInitialStylesOptionsFormatJsx): JSX.Element;
 export function getInitialStyles(opts?: GetInitialStylesOptionsWithoutTags): string;
 export function getInitialStyles(opts?: GetInitialStylesOptions): string | JSX.Element {
-  const { prefix, withoutTags, format }: GetInitialStylesOptions = {
+  const { prefix, withoutTags, format, normalize }: GetInitialStylesOptions = {
```
Comment (tatjana-gartner2) [inline packages/components-js/projects/partials/scripts/generateInitialStylesPartial.ts:110]: rename applyWithNormalizeStyles

File: packages/components-js/projects/partials/scripts/generateInitialStylesPartial.ts [hunk 8]
```diff
@@ -34,9 +125,11 @@ export function getInitialStyles(opts?: GetInitialStylesOptions): string | JSX.E
 
   const styles = prefixedTagNames.join() + '{visibility:hidden}.hydrated,.ssr{visibility:inherit}';
 
+  const normalizeStyles = \`${minifyCss(getCss(normalizeStyles))}\`;
+
   const markup = format === 'html'
-    ? \`<style \$\{styleAttributes\}>\${styles}</style>\`
-    : <style {...styleProps} dangerouslySetInnerHTML={{ __html: styles }} />;
+    ? \`<style \$\{styleAttributes\}>\${normalize ? styles.concat(normalizeStyles) : styles}</style>\`
```
Comment (tatjana-gartner2) [inline packages/components-js/projects/partials/scripts/generateInitialStylesPartial.ts:131]: use ternary only once

File: packages/components-js/projects/partials/tests/unit/specs/__snapshots__/getInitialStyles.spec.ts.snap [hunk 9]
```diff
@@ -0,0 +1,3 @@
...
...t}html{text-size-adjust:none;-webkit-text-size-adjust:none;font-family:'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif}button, input, optgroup, select, textarea{text-size-adjust:none;-webkit-text-size-adjust:none;font-family:'Porsche Next','Arial Narrow',Arial,'Heiti SC',SimHei,sans-serif;font-size:1rem;line-height:calc(6px + 2.125ex);font-weight:400;font-style:normal;font-variant:normal;overflow-wrap:break-word;hyphens:auto}a{color:inherit;text-decoration:underline}b, strong{font-weight:700}em, i{font-style:normal}a, button, input, select, textarea{outline:1px solid transparent;outline-offset:1px}a::-moz-focus-inner, button::-moz-focus-inner, input::-moz-focus-inner, select::-moz-focus-inner, textarea::-moz-focus-inner{border:0}a:focus, button:focus, input:focus, select:focus, textarea:focus{outline-color:currentColor}a:focus:not(:focus-visible), button:focus:not(:focus-visible), input:focus:not(:focus-visible), select:focus:not(:focus-visible), textarea:focus:not(:focus-visible){outline-color:transparent}@media(hover:hover){a{transition:color var(--p-transition-duration, .24s) ease}a:hover{color:#d5001c}[data-theme="dark"] a:hover:hover{color:#ff0223}}</style>"`;
```
Comment (tatjana-gartner2) [inline packages/components-js/projects/partials/tests/unit/specs/__snapshots__/getInitialStyles.spec.ts.snap:3]: remove whitespaces

File: packages/components-react/projects/react-ssr-wrapper/tests/unit/specs/__snapshots__/react-ssr-wrapper.spec.tsx.snap [hunk 10]
```diff
@@ -3092,12 +3092,12 @@ exports[`should render dsr component for PLinkPure 1`] = `
   vertical-align: top;
 }
 ::slotted(a) {
-  display: block;
-  position: static;
-  text-decoration: none;
-  font: inherit;
-  color: inherit;
-  outline: 0;
+  display: block !important;
```
Comment (tatjana-gartner2) [inline packages/components-react/projects/react-ssr-wrapper/tests/unit/specs/__snapshots__/react-ssr-wrapper.spec.tsx.snap:3095]: add important to all slotted styles

File: packages/storefront/src/pages/partials/initial-styles.md [hunk 11]
```diff
@@ -11,11 +11,12 @@ which needs to be injected into the `<head>` of your `index.html`.
...
+| `prefix`      | Prefix will be added to the component names.                                                                                                                                                                                                  | `string`  | `''`    |
+| `withoutTags` | <span style='color:#d5001c'>**[DEPRECATED]**</span> since v2.9.0 and will be removed in v3, use `format: 'jsx'` instead.<br/>If true, it returns css styles.                                                                                  | `boolean` | `false` |
+| `format`      | Defines the output format of the partial. By default, it returns a html string, with `jsx` it returns valid jsx elements.                                                                                                                     | `'html'   | 'jsx'`  | `'html'` |
+| `normalize`   | Applies basic css styles for Light DOM (Focus style on focusable elements, font styles on form elements, basic styles and hover styles on anchor elements, proper font weight on bold elements, and correct font style on emphasis elements). | `boolean` | `false` |
```
Comment (tatjana-gartner2) [inline packages/storefront/src/pages/partials/initial-styles.md:19]: edd to example

File: packages/components-js/projects/partials/scripts/generateInitialStylesPartial.ts [hunk 12]
```diff
@@ -8,19 +11,79 @@ export const generateInitialStylesPartial = (): string => {
...
+          transition: 'color var(--p-transition-duration, .24s) ease',
+          '&:hover': {
+            color: themeLight.state.hover,
+          },
+        },
+      },
+
+      '[data-theme="dark"] a:hover': {
+        '@media(hover:hover)': {
+          '&:hover': {
```
Comment (denyo) [inline packages/components-js/projects/partials/scripts/generateInitialStylesPartial.ts:51]: This produces `[data-theme="dark"] a:hover:hover{color:#ff0223}` which doesn't seem right.

File: packages/components-js/projects/partials/scripts/generateInitialStylesPartial.ts [hunk 13]
```diff
@@ -1,5 +1,8 @@
 import { joinArrayElementsToString, withoutTagsOption } from './utils';
-import { INTERNAL_TAG_NAMES, TAG_NAMES } from '@porsche-design-system/shared';
+import { INTERNAL_TAG_NAMES, TAG_NAMES, getMinifiedCss } from '@porsche-design-system/shared';
+import { Styles } from 'jss';
+import { themeDark, fontWeight, themeLight, textSmall } from '@porsche-design-system/utilities-v2';
+import { fontBehavior, fontFamily, fontHyphenation, fontLineHeight } from '@porsche-design-system/utilities-v2/src/js';
```
Comment (denyo) [inline packages/components-js/projects/partials/scripts/generateInitialStylesPartial.ts:5]: Why not import from built package together with imports 1 line above?

File: packages/components-js/projects/partials/scripts/generateInitialStylesPartial.ts [hunk 14]
```diff
@@ -8,19 +11,79 @@ export const generateInitialStylesPartial = (): string => {
   prefix?: string;
   ${withoutTagsOption}
   format?: Format;
+  // Standard normalize.css styles won't be applied by using this option
+  applyWithNormalizeStyles?: boolean;
```
Comment (denyo) [inline packages/components-js/projects/partials/scripts/generateInitialStylesPartial.ts:15]: I am not sure about the wording. Sounds weird and unnatural.

File: packages/components-js/projects/partials/scripts/generateInitialStylesPartial.ts [hunk 15]
```diff
@@ -46,7 +53,7 @@ type GetInitialStylesOptionsWithoutTags = Omit<GetInitialStylesOptions, 'format'
         },
       },
 
-      '[data-theme="dark"] a:hover': {
+      'a[data-theme="dark"]': {
```
Comment (denyo) [inline packages/components-js/projects/partials/scripts/generateInitialStylesPartial.ts:56]: Now it produces `a[data-theme="dark"]:hover{color:#ff0223}}` which isn't right, either?

**Review by denyo — APPROVED** [review]

**Pull request merged**